tools/mage/deploy_frontend.go

```
@@ -0,0 +1,92 @@
+package mage
+
+/**
+ * Panther is a scalable, powerful, cloud-native SIEM written in Golang/React.
+ * Copyright (C) 2020 Panther Labs Inc
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/fatih/color"
+	"github.com/magefile/mage/sh"
+)
+
+// Functions that build a personalized docker image from source, while pushing it to the private image repo of the user
+func buildAndPushImageFromSource(awsSession *session.Session, imageTag string) error {
+	fmt.Println("deploy: Requesting access to remote image repo")
+	ecrClient := ecr.New(awsSession)
+	req, resp := ecrClient.GetAuthorizationTokenRequest(&ecr.GetAuthorizationTokenInput{})
+	if err := req.Send(); err != nil {
+		return err
+	}
+
+	ecrAuthorizationToken := *resp.AuthorizationData[0].AuthorizationToken
+	ecrServer := *resp.AuthorizationData[0].ProxyEndpoint
+
+	decodedCredentialsInBytes, _ := base64.StdEncoding.DecodeString(ecrAuthorizationToken)
```

Never ignore errors! This should be

```go
decodedCredentialsInBytes, err := base64.StdEncoding.DecodeString(ecrAuthorizationToken)
if err != nil {
    return err
}
```

---

ook